packages/flutter/lib/src/gestures/multitap.dart

```diff
@@ -152,23 +169,45 @@ class DoubleTapGestureRecognizer extends GestureRecognizer {
   ///  * [kPrimaryButton], the button this callback responds to.
   GestureDoubleTapCallback onDoubleTap;
 
+  /// Called when a pointer that will trigger a tap of a primary button has
+  /// stopped contacting the screen at a particular location, unless a previous
+  /// tap landed in proximity shortly before, in which case [onDoubleTap] and
+  /// [onDoubleTapUp] will be called.
+  ///
+  /// [onFirstTapUp] is called optimistically before winning the gesture arena,
+  /// the same pointer that triggered [onFirstTapUp] might be rejected afterwards.
+  /// Neither [onDoubleTap] or [onDoubleTapUp] is guaranteed to be called after
+  /// [onFirstTapUp].
+  GestureTapUpCallback onFirstTapUp;
+
```

Comment:
I think we'll need a cancel callback. Otherwise, if users of this API cached something from the first tap up callback, the user would never know when what was cached becomes obsolete.